Game crashes when trying to pick up Dragunov via merc directly or when trying to open map screen inventory if there is a dragunov on the map

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  v0.17.0 (2020-08-03)| JA 2 english 1.12 (GOG)  |    ?   |    ?    |
also using all the the mods included in the v0.17.0 version

using Windows 10

Game crashes when trying to pick up a Dragunov via a merc directly on the map or when trying to open the map screen inventory if there is a dragunov on the map. It happens in Alma H13. It`s teh first time that the oppenents dropped a dragunov and it happens with both seperate Dragunovs on the map.
Maybe the reason is a problem with dragunov skin in the Kaerar gun pack ?


2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: ------------------------------------------------------------------------------
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: JA2 Home Dir:                  'C:\Users\Moritz\Documents\JA2'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: Root game resources directory: 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: Extra data directory:          'D:\System\JA2 Stracciatella'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: Data directory:                'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: Tilecache directory:           'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\tilecache'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: Saved games directory:         'C:\Users\Moritz\Documents\JA2\SavedGames-from-russia-with-love-generous-rebels-imp-quiz-honest-answers-kaerar-gun-pack-nightops-maps-o-fortuna-playable-characters-mod-test-json-dialogs'
2021-01-13T18:38:40 [ INFO] sgp\SGP.cc: ------------------------------------------------------------------------------
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\from-russia-with-love\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'from-russia-with-love'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\generous-rebels\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'generous-rebels'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\imp-quiz-honest-answers\data'2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'imp-quiz-honest-answers'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\kaerar-gun-pack\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'kaerar-gun-pack'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\nightops-maps\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'nightops-maps'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\o-fortuna\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'o-fortuna'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\playable-characters-mod\data'2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'playable-characters-mod'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (100) 'D:\System\JA2 Stracciatella\mods\test-json-dialogs\data'
2021-01-13T18:38:40 [ INFO] externalized\ModPackContentManager.cc: Loaded mod 'test-json-dialogs'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Skipped adding to VFS 'C:\Users\Moritz\Documents\JA2\data': Path does not exist
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (210) 'D:\System\JA2 Stracciatella\externalized'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding directory to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Ambient.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Anims.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\BattleSNDS.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Bigitems.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Binarydata.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Cursors.slf'
2021-01-13T18:38:40 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Data.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Faces.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Fonts.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Interface.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Intro.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Laptop.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Loadscreens.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Maps.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Mercedt.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Music.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Npcdata.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Npc_speech.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Radarmaps.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Sounds.slf'
2021-01-13T18:38:41 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Speech.slf'
2021-01-13T18:38:42 [ INFO] externalized\DefaultContentManager.cc: Adding SLF archive to VFS (220) 'C:\Program Files (x86)\GOG Galaxy\Games\Jagged Alliance 2\data\Tilesets.slf'
2021-01-13T18:38:44 [ INFO] game\GameLoop.cc: Version Label: Stracciatella v0.17.0+1e6667a
2021-01-13T18:38:44 [ INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2021-01-13T18:39:04 [ WARN] game\TileEngine\Ambient_Control.cc: Cannot load Ambient data for tileset
2021-01-13T18:39:05 [ WARN] game\Tactical\Tactical_Save.cc: SynchronizeItemTempFile() Reported 0, should be 11
2021-01-13T18:39:27 [ERROR] launcher\Launcher.cc: Command_execute "D:\\System\\JA2 Stracciatella\\ja2.exe" []: exited with status code -1073741819
![Rust error](https://user-images.githubusercontent.com/77403435/104495131-a6d15b80-55d7-11eb-8b32-99056990d552.JPG)


Thanks!

--- a/src/game/Tactical/LoadSaveObjectType.cc
+++ b/src/game/Tactical/LoadSaveObjectType.cc
@@ -7,6 +7,16 @@
 #include "ItemModel.h"
 
 
+static void ReplaceInvalidItem(UINT16 & usItem)
+{
+	const ItemModel* item = GCM->getItem(usItem);
+	if (!item)
+	{
+		STLOGW("Item (index {}) is not defined and will be ignored. Maybe the file was saved for a different game version", usItem);
+		usItem = NONE;
+	}
+}
+
 void ExtractObject(DataReader& d, OBJECTTYPE* const o)
 {
 	size_t start = d.getConsumed();
@@ -14,13 +24,8 @@ void ExtractObject(DataReader& d, OBJECTTYPE* const o)
 	EXTR_U8(d, o->ubNumberOfObjects)
 	EXTR_SKIP(d, 1)
 
+	ReplaceInvalidItem(o->usItem);
 	const ItemModel* item = GCM->getItem(o->usItem);
-	if (!item)
-	{
-		STLOGW("Item (index {}) is not defined and will be ignored. Maybe the file was saved for a different game version", o->usItem);
-		item = GCM->getItem(NONE);
-	}
-
 	switch (item->getItemClass())
 	{
 		case IC_AMMO:
@@ -106,6 +111,12 @@ extract_status:
 	EXTR_U8(d, o->fUsed)
 	EXTR_SKIP(d, 2)
 	Assert(d.getConsumed() == start + 36);
+
+	// Check and remove invalid items in attachment slots
+	for (UINT16 & i : o->usAttachItem)
+	{
+		ReplaceInvalidItem(i);
+	}
 }
 
 

